grpc-proto/src/main/java/io/stargate/grpc/Values.java

```diff
@@ -163,4 +165,143 @@ public static Value of(Map<Value, Value> fields) {
+    if (value.getInnerCase() != InnerCase.INT) {
+      throw new IllegalArgumentException("Expected int value");
+    }
+    int intValue = (int) value.getInt();
+    if (intValue != value.getInt()) {
+      throw new IllegalArgumentException(
+          String.format("Valid range for int is %d to %d", Integer.MIN_VALUE, Integer.MAX_VALUE));
+    }
+    return intValue;
+  }
+
+  public static long bigint(Value value) {
+    if (value.getInnerCase() != InnerCase.INT) {
+      throw new IllegalArgumentException("Expected int value");
+    }
+    return value.getInt();
+  }
+
+  public static short smallint(Value value) {
+    if (value.getInnerCase() != InnerCase.INT) {
+      throw new IllegalArgumentException("Expected int value");
+    }
+    short shortValue = (short) value.getInt();
+    if (shortValue != value.getInt()) {
+      throw new IllegalArgumentException(
+          String.format("Valid range for int is %d to %d", Short.MIN_VALUE, Short.MAX_VALUE));
+    }
+    return shortValue;
+  }
+
+  public static byte tinyint(Value value) {
+    if (value.getInnerCase() != InnerCase.INT) {
+      throw new IllegalArgumentException("Expected int value");
+    }
+    byte byteValue = (byte) value.getInt();
+    if (byteValue != value.getInt()) {
+      throw new IllegalArgumentException(
+          String.format("Valid range for int is %d to %d", Byte.MIN_VALUE, Byte.MAX_VALUE));
+    }
+    return byteValue;
+  }
+
+  public static float float_(Value value) {
+    if (value.getInnerCase() != InnerCase.FLOAT) {
+      throw new IllegalArgumentException("Expected float value");
+    }
+    return value.getFloat();
+  }
+
+  public static double double_(Value value) {
+    if (value.getInnerCase() != InnerCase.DOUBLE) {
+      throw new IllegalArgumentException("Expected double value");
+    }
+    return value.getDouble();
+  }
+
+  public static ByteBuffer byteBuffer(Value value) {
+    if (value.getInnerCase() != InnerCase.BYTES) {
+      throw new IllegalArgumentException("Expected bytes value");
+    }
+    ByteBuffer bytes = ByteBuffer.allocate(value.getBytes().size());
+    value.getBytes().copyTo(bytes);
+    return bytes;
+  }
+
+  public static byte[] bytes(Value value) {
+    if (value.getInnerCase() != InnerCase.BYTES) {
+      throw new IllegalArgumentException("Expected bytes value");
+    }
+    return value.getBytes().toByteArray();
+  }
+
+  public static String string(Value value) {
+    if (value.getInnerCase() != InnerCase.STRING) {
+      throw new IllegalArgumentException("Expected string value");
+    }
+    return value.getString();
+  }
+
+  public static UUID uuid(Value value) {
+    if (value.getInnerCase() != InnerCase.UUID) {
+      throw new IllegalArgumentException("Expected uuid value");
+    }
+    ByteBuffer bytes = ByteBuffer.allocate(16);
+    value.getUuid().getValue().copyTo(bytes);
+    bytes.flip();
+    return new UUID(bytes.getLong(bytes.position()), bytes.getLong(bytes.position() + 8));
```

Comment:
Could just use positions `0` and `8`, since we allocated `ByteBuffer`.
That is, unless we could avoid copy altogether if `value` exposes `ByteBuffer` or allows read of 2 `long`s we need
(java-uuid-generator package would also have way to constructor `UUID` directly from `byte[]` but probably don't want to add a new dependency just for that).

Reply:
good catch. I wish that that `UUID(byte[])` constructor was public. Switch to using absolute positions.

Reply:
Right. Seems odd that constructor (or factory method) is not provided.